### DISCUSSION TITLE
Python flake8 and autopep8  dependency error

### DISCUSSION THREAD

**Discussion by zeeshanali1993** [intelowlproject/IntelOwl#1872] [Q&A]
### **Environment:**
**Ubuntu: 22.04.3
Python: 3.10.12
IntelOwl: 5.1.0**

flake8 and autopep8 dependency conflict as **flake8** requires  **pycodestyle>=2.8.0 <2.9.0**, whereas **autopep8** requires **pycodestyle>=2.9.1**. Now installed pycodestyle=2.9.0 tried to downgrade autopep8 from 2.0.4 to 2.0.0 but it is incompatible with the pycodestyle 2.8.0.
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. 
This behaviour is the source of the following dependency conflicts.
flake8 4.0.1 requires pycodestyle<2.9.0,>=2.8.0, but you have pycodestyle 2.9.0 which is incompatible.
autopep8 2.0.0 requires pycodestyle>=2.9.1, but you have pycodestyle 2.9.0 which is incompatible.
```

**Comment by mlodic** [comment]
hey, is `autopep8` installed by default? I think not. So that's normal that there could be version conflict if you use other packages.

Then, I understand that `flake8` is an old version so I have just updated it to the most recent version in the `develop` branch.
